smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy

```diff
@@ -629,15 +704,57 @@ def poll() {
 			pollAttempt = 3
 		}
 	}
+	markChildrenOffline()
 	log.trace "poll exit pollAttempt:${pollAttempt}, isThermostatPolled:${isThermostatPolled}, " +
 			"isSwitchesPolled:${isSwitchesPolled}"
 }
 
-def markChildrenOffline() {
-	def childDevices = getChildDevices()
-	childDevices.each{ child ->
-		child.sendEvent(name: "DeviceWatch-DeviceStatus", value: "offline", displayed: false)
-		child.sendEvent("name":"thermostat", "value":"Offline")
+def markChildrenOffline(boolean markAllOffline = false) {
+	if (markAllOffline) {
+		def childDevices = getChildDevices()
+		childDevices.each{ child ->
+			child.sendEvent(name: "DeviceWatch-DeviceStatus", value: "offline", displayed: false)
+			child.sendEvent("name":"thermostat", "value":"Offline")
+		}
+	} else {
+		def remoteSensors = state.remoteSensors2 ?: [:]
+		def thermostatList = state.thermostats ?: [:]
+		def switchList = state.switchList ?: [:] 
+		// For devices offline that wasn't polled update pollAttemps, if this is 3rd pollAttempts, mark device offline
+		def thermostatsOffline = thermostatList.findAll { dni, stat ->
+			if ((stat.data.deviceAlive == false) && (stat.polled == false)) {
+				stat.pollAttempts = stat.pollAttempts ? stat.pollAttempts + 1 : 1
+				if (stat.pollAttempts > 2) {
+					return dni
+				}
+			}
+		}?.keySet()
+		def remoteSensorsOffline = remoteSensors.findAll { rsdni, sensor ->
+			if ((sensor.deviceAlive == false) && (sensor.polled == false)) {
+				sensor.pollAttempts = sensor.pollAttempts ? sensor.pollAttempts + 1 : 1
+				if (sensor.pollAttempts > 2) {
+					return rsdni
+				}
+			}
+		}?.keySet()
+		def switchesOffline = switchList.findAll { sdni, sw ->
+			if ((sw.deviceAlive == false) && (sw.polled == false)) {
+				sw.pollAttempts = sw.pollAttempts ? sw.pollAttempts + 1 : 1
+				if (sw.pollAttempts > 2) {
+					return sdni
+				}
+			}
+		}?.keySet()
```

Comment:
It seems like all of the above three are doing the same thing. Should we move this out into a helper function?

Reply:
two are doing pretty much the same thing, the third does not, I didn't feel like sorting out the helper method for two things when the third was different 😪 Leaving that for another day....